src/chocolatey/GetChocolatey.cs

```diff
@@ -122,10 +128,25 @@ public class GetChocolatey
         /// Initializes a new instance of the <see cref="GetChocolatey"/> class.
         /// </summary>
         public GetChocolatey()
+            : this(true)
+        {
+        }
+
+        public GetChocolatey(bool initializeLogging)
         {
-            Log4NetAppenderConfiguration.configure(null, excludeLoggerNames: ChocolateyLoggers.Trace.to_string());
-            Bootstrap.initialize();
-            Log.InitializeWith(new AggregateLog(new List<ILog>() { new Log4NetLog(), _logSinkLogger }));
+            if (initializeLogging)
+            {
+                string loggingLocation = ApplicationParameters.LoggingLocation;
+                
+                if (!Directory.Exists(loggingLocation))

```

Comment:
I am just wondering about the direct usage of the Directory class here.  Normally, where possible we use the FileSystem abstraction when IO operations are performed.  While we don't have a direct IFileSystem instance here, we do have access to the `_container` where we could get one from.  Thoughts?

Reply:
I'm not sure, the main reason for using the `Directory` class here was due to Rob also did this himself in the Work he had done.
I only assumed it was because at this point chocolatey hadn't yet been set up, but I'll run some tests to check if it would be possible for us to use `IFileSystem` already here.

Reply:
I did some testing around using the `IFileSystem` resolved from the container, and that works as it should.

The PR have been updated to make use of it instead of raw calls to the `Directory` class.